common/common.cpp

```diff
@@ -28,7 +28,7 @@ mender::common::expected::ExpectedLongLong StringToLongLong(const string &str, i
 	if (errno != 0) {
 		int int_error = errno;
 		return mender::common::error::Error(
-			std::error_code(int_error, std::system_category()).default_error_condition(), "");
+			std::generic_category().default_error_condition(int_error), "");
```

Comment:
Are you able to differentiate between https://en.cppreference.com/w/cpp/error/generic_category and https://en.cppreference.com/w/cpp/error/system_category? They appear to be saying almost the same thing, and both use `errno` as example! Not very helpful documentation...

This looks good though, better than my original approach.

Reply:
It is very confusing. I actually go to this when searching for how to get errors from stream operation failures and it's mentioned [here](https://stackoverflow.com/a/66229636) with the following:

> `std::system_category()` will map the error codes from system's native error code facility. For *nix, this is `errno`. For Win32, it is `GetLastError()`.

So with `system_category` you should use `GetLastError()` on Windows (or whatever system-specific thing), `generic_category` works with `errno` everywhere. But it's confusing as hell, so...

Reply:
Aha! I guess that sort of makes sense, but it could definitely have been clearer. I guess we should stick to `generic_category` then, since even on Windows we would use the POSIX API as it's currently implemented.